mountpoint-s3-fs/tests/fuse_tests/metrics_test.rs

```diff
@@ -0,0 +1,226 @@
+use crate::common::fuse::{TestSessionConfig, mock_session, read_dir_to_entry_names};
+use crate::common::test_recorder::{Metric, TestRecorder};
+use mountpoint_s3_fs::S3FilesystemConfig;
+use mountpoint_s3_fs::metrics::defs::{
+    ATTR_FUSE_REQUEST, FUSE_IO_SIZE, FUSE_REQUEST_ERRORS, FUSE_REQUEST_LATENCY, PREFETCH_RESET_STATE,
+};
+use rusty_fork::rusty_fork_test;
+use std::fs::{File, read_dir};
+use std::io::{Read, Seek, Write};
+use std::sync::Arc;
+
+fn setup_recorder() -> TestRecorder {
+    let recorder = TestRecorder::default();
+    metrics::set_global_recorder(recorder.clone()).expect("Failed to set global recorder");
+    recorder
+}
+
+fn get_metric(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) -> Arc<Metric> {
+    recorder
+        .get(name, labels)
+        .unwrap_or_else(|| panic!("Expected metric '{name}' with labels {labels:?} to exist"))
+}
+
+fn get_histogram(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) -> Vec<f64> {
+    let metric = get_metric(recorder, name, labels);
+    match metric.as_ref() {
+        Metric::Histogram(_) => metric.histogram(),
+        _ => panic!("Metric '{name}' is not a histogram"),
+    }
+}
+
+fn assert_metric_exists(recorder: &TestRecorder, name: &str, labels: &[(&str, &str)]) {
+    let _ = get_metric(recorder, name, labels);
+}
+
+fn verify_common_metrics(recorder: &TestRecorder) {
+    for request in ["open", "getattr", "fsync"] {
+        assert_metric_exists(recorder, FUSE_REQUEST_LATENCY, &[(ATTR_FUSE_REQUEST, request)]);
+    }
+    assert_metric_exists(recorder, "fs.inodes", &[]);
+    assert_metric_exists(recorder, "fs.inode_kinds", &[("kind", "file")]);
+    assert_metric_exists(recorder, "fs.inode_kinds", &[("kind", "directory")]);
+    assert_metric_exists(recorder, "fuse.op_unimplemented", &[("op", "getxattr")]);
+
+    assert_metric_exists(recorder, "fuse.total_threads", &[]);
+    assert_metric_exists(recorder, "fuse.idle_threads", &[]);
+}
+
+rusty_fork::rusty_fork_test! {
```

Comment:
nit: `rusty_fork_test` is already imported, so `rusty_fork::` should not be necessary